go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -51,6 +51,31 @@ type PluginProperties struct {
 	// override that behavior unless the resource that gets created for this plugin does not consume resources (cluster's
 	// cpu/memory... etc. or external resources) once the plugin's Plugin.GetTaskPhase() returns a terminal phase.
 	DisableDeleteResourceOnFinalize bool
+	// Optionally specifies an override for the default OnAbort behavior (deleting the underlying resource).
+	AbortOverride PluginAbortOverride
+}
+
+type PluginAbortOverride interface {
+	OnAbort(ctx context.Context, tCtx pluginsCore.TaskExecutionContext, resource client.Object) (behavior AbortBehavior, err error)
+}
+
+type AbortBehavior struct {
+	// Optional override to the default k8s Resource being acted on.
+	Resource       client.Object
+	DeleteResource bool
```

Comment:
an additional thought is to add either custom constructors here, or a check in flytepropeller, to make sure that, for example, users don't set both `Delete` and `Update`, or other invalid configurations.

Reply:
Excellent point! I like the custom constructors pattern yes... 
e.g. https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/core/phase.go#L149-L220

Reply:
ok! I added some custom constructors in that style. It's a lot of LOC, not sure if I should move them into a separate file at this point?